src/uu/lsmem/src/lsmem.rs

```diff
@@ -68,75 +58,128 @@ enum Columns {
     #[serde(rename = "ZONES")]
     Zones,
 }
-// const SCOLS_FL_RIGHT: i32 = 1;
-
-// static COLDESCS: [ColDesc; 7] = [
-//     ColDesc {
-//         name: "RANGE",
-//         whint: 0.0,
-//         flags: 0,
-//         help: "start and end address of the memory range",
-//     },
-//     ColDesc {
-//         name: "SIZE",
-//         whint: 5.0,
-//         flags: SCOLS_FL_RIGHT,
-//         help: "size of the memory range",
-//     },
-//     ColDesc {
-//         name: "STATE",
-//         whint: 0.0,
-//         flags: SCOLS_FL_RIGHT,
-//         help: "online status of the memory range",
-//     },
-//     ColDesc {
-//         name: "REMOVABLE",
-//         whint: 0.0,
-//         flags: SCOLS_FL_RIGHT,
-//         help: "memory is removable",
-//     },
-//     ColDesc {
-//         name: "BLOCK",
-//         whint: 0.0,
-//         flags: SCOLS_FL_RIGHT,
-//         help: "memory block number or blocks range",
-//     },
-//     ColDesc {
-//         name: "NODE",
-//         whint: 0.0,
-//         flags: SCOLS_FL_RIGHT,
-//         help: "numa node of memory",
-//     },
-//     ColDesc {
-//         name: "ZONES",
-//         whint: 0.0,
-//         flags: SCOLS_FL_RIGHT,
-//         help: "valid zones for the memory range",
-//     },
-// ];
-
-#[derive(Debug, Deserialize, PartialEq, Clone, Copy)]
+
+impl ValueEnum for Column {
+    fn value_variants<'a>() -> &'a [Self] {
+        &[
+            Column::Range,
+            Column::Size,
+            Column::State,
+            Column::Removable,
+            Column::Block,
+            Column::Node,
+            Column::Zones,
+        ]
+    }
+
+    fn to_possible_value(&self) -> Option<PossibleValue> {
+        Some(PossibleValue::new(self.get_name()))
+    }
+}
+
+/// Default columns to display if none are explicitly specified.
+const DEFAULT_COLUMNS: &[Column] = &[
+    Column::Range,
+    Column::Size,
+    Column::State,
+    Column::Removable,
+    Column::Block,
+];
+/// Which columns (attributes) are possible to split memory blocks to ranges on.
+const SPLIT_COLUMNS: &[Column] = &[
+    Column::State,
+    Column::Removable,
+    Column::Node,
+    Column::Zones,
+];
+
+impl Column {
+    fn get_name(&self) -> &'static str {
+        match self {
+            Column::Range => "RANGE",
+            Column::Size => "SIZE",
+            Column::State => "STATE",
+            Column::Removable => "REMOVABLE",
+            Column::Block => "BLOCK",
+            Column::Node => "NODE",
+            Column::Zones => "ZONES",
+        }
+    }
+
+    #[allow(dead_code)]
+    fn get_float_right(&self) -> bool {
+        self != &Column::Range
+    }
+
+    #[allow(dead_code)]
```

Comment:
A detail: using `#[allow(dead_code)]` isn't necessary as both methods are used in the `print_table` function.
```suggestion
    fn get_float_right(&self) -> bool {
        self != &Column::Range
    }

```